crates/bgp-pkt/src/bgp_ls.rs

```diff
@@ -0,0 +1,1273 @@
+            }
+            BgpLsAttributeTlv::MplsProtocolMask { .. } => iana::BgpLsAttributeTlv::MplsProtocolMask,
+            BgpLsAttributeTlv::IgpMetric(..) => iana::BgpLsAttributeTlv::IgpMetric,
+            BgpLsAttributeTlv::SharedRiskLinkGroup(..) => {
+                iana::BgpLsAttributeTlv::SharedRiskLinkGroup
+            }
+            BgpLsAttributeTlv::OpaqueLinkAttribute(..) => {
+                iana::BgpLsAttributeTlv::OpaqueLinkAttribute
+            }
+            BgpLsAttributeTlv::LinkName(..) => iana::BgpLsAttributeTlv::LinkName,
+            BgpLsAttributeTlv::IgpFlags { .. } => iana::BgpLsAttributeTlv::IgpFlags,
+            BgpLsAttributeTlv::IgpRouteTag(_) => iana::BgpLsAttributeTlv::IgpRouteTag,
+            BgpLsAttributeTlv::IgpExtendedRouteTag(_) => {
+                iana::BgpLsAttributeTlv::IgpExtendedRouteTag
+            }
+            BgpLsAttributeTlv::PrefixMetric(_) => iana::BgpLsAttributeTlv::PrefixMetric,
+            BgpLsAttributeTlv::OspfForwardingAddress(_) => {
+                iana::BgpLsAttributeTlv::OspfForwardingAddress
+            }
+            BgpLsAttributeTlv::OpaquePrefixAttribute(_) => {
+                iana::BgpLsAttributeTlv::OpaquePrefixAttribute
+            }
+            BgpLsAttributeTlv::MultiTopologyIdentifier(..) => {
+                iana::BgpLsAttributeTlv::MultiTopologyIdentifier
+            }
+            BgpLsAttributeTlv::NodeFlagBits { .. } => iana::BgpLsAttributeTlv::NodeFlagBits,
+            BgpLsAttributeTlv::OpaqueNodeAttribute(..) => {
+                iana::BgpLsAttributeTlv::OpaqueNodeAttribute
+            }
+            BgpLsAttributeTlv::NodeNameTlv(..) => iana::BgpLsAttributeTlv::NodeNameTlv,
+            BgpLsAttributeTlv::IsIsArea(..) => iana::BgpLsAttributeTlv::IsIsArea,
+            BgpLsAttributeTlv::PeerNodeSid(..) => iana::BgpLsAttributeTlv::PeerNodeSid,
+            BgpLsAttributeTlv::PeerAdjSid(..) => iana::BgpLsAttributeTlv::PeerAdjSid,
+            BgpLsAttributeTlv::PeerSetSid(..) => iana::BgpLsAttributeTlv::PeerSetSid,
+        }
+    }
+}
+
+/// ```text
+///  0                   1                   2                   3
+///  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |            NLRI Type          |     Total NLRI Length         |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |                                                               |
+/// //                  Link-State NLRI (variable)                 //
+/// |                                                               |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// ```
+/// see [RFC7752 Section 3.2](https://www.rfc-editor.org/rfc/rfc7752#section-3.2)
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(feature = "fuzz", derive(arbitrary::Arbitrary))]
+pub struct BgpLsNlri {
+    pub path_id: Option<u32>,
```

Comment:
I'm not sure, does any of the RFCs define that BGP LS NLRI can have multiple paths?